### ISSUE TITLE
 Why is my avatar squatting?

### ISSUE THREAD

**Issue by BogusCurry** [WhiteCoreSim/WhiteCore-Dev#239]
Hello ;D

Why is my avatar squatting?
![snapshot_001](https://cloud.githubusercontent.com/assets/614576/20951895/f0881ae0-bc28-11e6-95fa-8f15c2fc847c.png)

aktual WCS dev
Linux Mint 18
Singu 1.86

Have a nice Day ;D

**Comment by emperorstarfinder** [comment]
I have seen this one myself actually but haven't had a chance to look into it yet.  Can you provide any additional info as to what the console is saying about the avatar if any at all? (you can also set console log level to debug which will provide debugging info that is not seen in log level info)

ALso have you tried moving the avatar around and if so does it do this after the avatar moves?

**Comment by BogusCurry** [comment]
I use ODE physics engine, the default avatar has not this problem. But the avatar go over the terrain, not on  the terrain.

**Comment by emperorstarfinder** [comment]
Great.  can you please test this with Bullet as well to see if you get this when using Bullet?

**Comment by BogusCurry** [comment]
Sorry, with Bulletsim is that bullshit .. no right jump ..when I fly and then makes fly off, is the terrain terrible .. I see nothing.

**Comment by emperorstarfinder** [comment]
I have done further testing on this and this has been confirmed.  There are no apparent error messages on the consoles to give any suggestions on possible issue.  Tested using ODE on Debian 8, Ubuntu 14.04 LTS, Ubuntu 16.04 LTS, Linux Mint 18.  Bullet will be the next test but safe to say this is likely a bug.

**Comment by Raider73** [comment]
I remember getting this issue back in the days of Aurora-Sim the fix was as simple as changing one of the settings in a ini file

**Comment by BogusCurry** [comment]
> I remember getting this issue back in the days of Aurora-Sim the fix was as simple as changing one of the settings in a ini file

In which ini file ?

**Comment by Raider73** [comment]
From memory something to do with the gravity settings

**Comment by emperorstarfinder** [comment]
That would be in the OpenDynamicsEngine.ini files.  However I recommend you try with the latest master as there has been a potential fix for the avatar height (scale).  It may be that the ODE was not accounting for the Real Avatar Height.

**Comment by greythane** [comment]
I have not seen this occurring in any testing that I have done, however...
The latest Singularity has a new 'Crouch toggle' key shortcut that may be causing this behaviour.

From http://www.singularityviewer.org/about
>>>
Area search
Windlight editor
Built in pose stand
[NEW] Crouch toggle with Shift-[Crouch button]
[NEW] Singularity -> Streaming Audio Display, find out what song is playing.
Alternatively: Vanity->General->Announce music stream's metadata in local chat when tuned in
<<<

This does not appear in the 6156 build (OSX) but I'm fairly sure that I have seen it in the latest Windows 6861 build (I was running the x64 build at the time)

**Comment by emperorstarfinder** [comment]
It happens actually in Firestorm as well so its not viewer specific.  Though I suspect its probably that Real Height bit I still need to test

**Comment by fly-man-** [comment]
Also, it might be that Hover Height isn't working properly in Whitecore as not all the bits and pieces for that have been implemented. So that might also give the "wrong" feeling when the AV is clearly standing on land

**Comment by greythane** [comment]
Bogus,
Can you give me some details on how you produced this please?
It looks like clean region but with your avatar modified.  
What is the reported height of your avi?

**Comment by emperorstarfinder** [comment]
My team's physics engine developer has found the problem and should have a more permanent fix for this shortly.  We will pass the fix to greythane to add to WC.  It has to do with the hover height and not ODE.

@BogusCurry if you adjust the hover height of your avatar to 0.168 your avatar should stand as normal.

**Comment by BogusCurry** [comment]
Hi there ;D
At first Happy New Year for all ;D

OK, I have use the Bulletsim, this in the date from bulletphysik.ini
http://paste.md-5.net/radepewece.ini
@emperorstarfinder What mean you with hover height from the avatar ? Is that AvatarCapsuleHeight ?

Greetings

Bogus | Found of GridTalk.de

**Comment by emperorstarfinder** [comment]
If you right click your avatar and go to appearance you will see an option that says Edit Hover height.  Change this to 0.168 and your avatar will be standing as normal.  The default Hover height for an avatar is 0.000 but there  is an issue tht is causing the crouching due to some pieces of the Hover height code not being implemented.

In most cases the default settings in the Bulletsim ini configuration files should be just fine.  All that would be necessary is just enabling bullet in Physics.ini

**Comment by BogusCurry** [comment]
OK, I have two regions, region one with bullet, region 2 with ode, when I log in on region 1 I can not crossover to region 2. That is very scary, when I log in on region 2, so I can crossover to region 1. 

The height from my avatar is now right, I not deep in the terrain.

**Comment by greythane** [comment]
Hi Bogus,
Regions set as BulletSim Physics do not currently allow region crossings.  This has always been the case as far as I am aware and has been inherited from the OpenSim code base.
I do have this 'on my list' to correct.

Regions crossings are fine using the ODE physics so that is why you an cross from your region 2.

**Comment by BogusCurry** [comment]
Hi Greythane ;D
Ok ;D Thanks for your answer and Happy New Year ;D

**Comment by emperorstarfinder** [comment]
At the time Aurora-Sim was forked from Opensim (0.6.3 - Opensim) Bulletsim was not the engine of choice for physics and was not really maintained.  So it is possible there are pieces inherited that would still cause a few issues for infinite regions and region crossings.  Region crossings with bulletsim do work but bullet doesn't store the information during the actual crossing correctly.  per the discussion @greythane and I have been having.

The reason I recommended testing the bulletsim was to see if the hover height was corrected for Bogus.  :)
